lib/node_modules/@stdlib/math/base/special/log10/src/main.c

```diff
@@ -160,56 +69,81 @@ static double klog( const double x ) {
 * // returns ~0.602
 */
 double stdlib_base_log10( const double x ) {
-	int32_t ihx;
-	int32_t ilx;
+	double valLo;
+	double valHi;
 	uint32_t hx;
 	uint32_t lx;
+	double hfsq;
+	double hi;
 	int32_t i;
 	int32_t k;
-	double xc;
-	double hi;
 	double lo;
+	double xc;
+	double y2;
 	double f;
+	double R;
+	double w;
 	double y;
-	double z;
 
+	if ( stdlib_base_is_nan( x ) || x < 0.0 ) {
+		return 0.0 / 0.0; // NaN
+	}
 	xc = x;
-	stdlib_base_float64_to_words( x, &hx, &lx );
-	ihx = (int32_t)hx;
-	ilx = (int32_t)lx;
+	stdlib_base_float64_to_words( xc, &hx, &lx );
 	k = 0;
-
-	// Case: 0 < x < 2**-1022
-	if ( ihx < HIGH_MIN_NORMAL_EXP ) {
-		if ( ( ( ihx & STDLIB_CONSTANT_FLOAT64_HIGH_WORD_ABS_MASK ) | ilx ) == 0 ) {
+	if ( hx < HIGH_MIN_NORMAL_EXP ) {
+		// Case: x < 2**-1022
+		if ( ( ( hx & STDLIB_CONSTANT_FLOAT64_HIGH_WORD_ABS_MASK ) | lx ) == 0 ) {
 			return STDLIB_CONSTANT_FLOAT64_NINF;
 		}
-		if ( ihx < 0 ) {
-			return 0.0 / 0.0; // NaN
-		}
-		// Subnormal number, scale up `x`...
 		k -= 54;
+
+		// Subnormal number, scale up x:
 		xc *= TWO54;
 		stdlib_base_float64_get_high_word( xc, &hx );
-		ihx = (int32_t)hx;
 	}
-	if ( ihx >= HIGH_MAX_NORMAL_EXP ) {
-		return x + x;
+	if ( hx >= HIGH_MAX_NORMAL_EXP ) {
+		return xc + xc;
+	}
+	// Case: log(1) = +0
+	if ( hx == HIGH_BIASED_EXP_0 && lx == 0 ) {
+		return 0;
 	}
-	k += ( ( ihx>>20 ) - STDLIB_CONSTANT_FLOAT64_EXPONENT_BIAS );
-	ihx &= STDLIB_CONSTANT_FLOAT64_HIGH_WORD_SIGNIFICAND_MASK;
-	i = ( ihx + 0x95f64 ) & 0x100000;
+	k += ( ( hx >> 20 ) - STDLIB_CONSTANT_FLOAT64_EXPONENT_BIAS );
+	hx &= STDLIB_CONSTANT_FLOAT64_HIGH_WORD_SIGNIFICAND_MASK;
+	i = ( hx + 0x95f64 ) & HIGH_MIN_NORMAL_EXP;
 
-	// Normalize `x` or `x/2`...
-	stdlib_base_float64_set_high_word( ihx | ( i ^ HIGH_BIASED_EXP_0 ), &xc );
-	k += i >> 20;
+	// Normalize x or x/2...
+	stdlib_base_float64_set_high_word( hx | ( i ^ HIGH_BIASED_EXP_0 ), &xc );
+	k += ( i >> 20 );
 	y = (double)k;
-	f = klog( xc );
-	xc -= 1.0;
-	hi = xc;
+	f = xc - 1.0;
+	hfsq = 0.5 * f * f;
+	R = stdlib_base_kernel_log1p( f );
+
+	/*
+	* Notes:
+	*
+	* -   `f-hfsq` must (for args near `1`) be evaluated in extra precision to avoid a large cancellation when `x` is near `sqrt(2)` or `1/sqrt(2)`.This is fairly efficient since `f-hfsq` only depends on `f`, so can be evaluated in parallel with `R`. Not combining `hfsq` with `R` also keeps `R` small (though not as small as a true `lo` term would be), so that extra precision is not needed for terms involving `R`.
```

Comment:
Can we add the missing space here (and potentially elsewhere where we have a similar comment?)

```suggestion
	* -   `f-hfsq` must (for args near `1`) be evaluated in extra precision to avoid a large cancellation when `x` is near `sqrt(2)` or `1/sqrt(2)`. This is fairly efficient since `f-hfsq` only depends on `f`, so can be evaluated in parallel with `R`. Not combining `hfsq` with `R` also keeps `R` small (though not as small as a true `lo` term would be), so that extra precision is not needed for terms involving `R`.
```

Reply:
Sure. Changed this in both `main.c` and `main.js`.